### ISSUE TITLE
OMEdit does not inherit Diagram and Icon size

### ISSUE THREAD

**Issue by mwetter** [OpenModelica/OpenModelica#10152]
### Description

When extending a model, OMEdit appears to ignore the Diagram and Icon size of the extended class.

### Steps to Reproduce
Loading the model below in OMEdit shows that `A` has a Diagram and Icon size that is too small (as it uses the default size).

```modelica
package TestGraphics

model A
  extends B;
end A;

model B
  annotation (
    Diagram(
      coordinateSystem(extent={{-200,-50},{200,50}}),
      graphics={
        Rectangle(
          extent={{-180,40},{180,-40}},
          lineColor={28,108,200})}),
    Icon(coordinateSystem(extent={{-200,-60},{140,100}}),
      graphics={Rectangle(
        extent={{-188,92},{130,-52}},
        lineColor={28,108,200},
        fillColor={0,140,72},
        fillPattern=FillPattern.Solid)}));
end B;

end TestGraphics;
```

### Expected Behavior
The two images should be the same:
![image](https://user-images.githubusercontent.com/1329080/216728933-a94d7cb1-e1b2-44a5-b6df-e11f6ae6ee1c.png)

![image](https://user-images.githubusercontent.com/1329080/216728801-93a5815d-977f-484a-a735-bca507d490a6.png)

This renders models such as `Buildings.Examples.Tutorial.CDL.System1` incorrectly.

### Related rendering issue

This also seems to be the reason why `Buildings.Experimental.DHC.Examples.Cooling.ElectricChillersDirectETS` renders the instance `dis` in a stretched way as shown below (on Buildings master, commit 21859cb56cabcdf004503ab9fd4d622ce80f2f3e).

![image](https://user-images.githubusercontent.com/1329080/216729737-38743168-332c-4430-9921-b368970486ce.png)

Changing
```diff
$ git diff
diff --git a/Buildings/Experimental/DHC/Examples/Cooling/ElectricChillersDirectETS.mo b/Buildings/Experimental/DHC/Examples/Cooling/ElectricChillersDirectETS.mo
index 2625656efd..1a607a261c 100644
--- a/Buildings/Experimental/DHC/Examples/Cooling/ElectricChillersDirectETS.mo
+++ b/Buildings/Experimental/DHC/Examples/Cooling/ElectricChillersDirectETS.mo
@@ -94,7 +94,7 @@ model ElectricChillersDirectETS "Example model for district cooling system with
   Modelica.Blocks.Sources.Constant TCHWSupSet(k=273.15+7)
     "Chilled water supply temperature setpoint"
     annotation (Placement(transformation(extent={{0,-60},{20,-40}})));
-  Buildings.Experimental.DHC.Networks.Distribution2PipePlugFlow dis(
+  Buildings.Experimental.DHC.Networks.BaseClasses.PartialDistribution dis(
     redeclare final package Medium=Medium,
     nCon=nLoa,
     allowFlowReversal=false,
```
which causes to use the model that declares the icon, rather than using a model that extends from the model that declares the icon, makes the diagram look fine.

### Version and OS

 - OpenModelica Version: OpenModelica 1.21.0~dev-220-g033f547
 - OS: Linux

**Comment by mwetter** [comment]
@casella : Test test with model `A` and `B` renders correctly if `--NAPI=true` is used. 
However, if one more level of hierarchy is introduced, it renders incorrectly. This test model renders incorrectly:

```modelica
package TestGraphics

model A
  extends B;
end A;

model B
  extends C;
end B;

model C
  annotation (
    Diagram(
      coordinateSystem(extent={{-200,-50},{200,50}}),
      graphics={
        Rectangle(
          extent={{-180,40},{180,-40}},
          lineColor={28,108,200})}),
    Icon(coordinateSystem(extent={{-200,-60},{140,100}}),
      graphics={Rectangle(
        extent={{-188,92},{130,-52}},
        lineColor={28,108,200},
        fillColor={0,140,72},
        fillPattern=FillPattern.Solid)}));
end C;

end TestGraphics;
```

Similarly, consistently with the new test model above,  `Buildings.Experimental.DHC.Examples.Cooling.ElectricChillersDirectETS` still has the stretched icon.

**Comment by casella** [comment]
OK, then @perost will have a look after the OpenModelica workshop.

**Comment by perost** [comment]
This looks like an OMEdit issue, the API returns the icon as expected.

**Comment by abuntrock** [comment]
@adeas31 With the OMEdit 1.22.0, which is the last stable release, this problems appears again as described in the original post by @mwetter. But with the newer nightly build v1.23.0-dev-168-gbc70ded451 (64-bit) it is solved again. Can you confirm this and will this fix be issue be solved in 1.22.1?

**Comment by casella** [comment]
@abuntrock 1.22.1 is out. Please check that. If the problem is still there, please open another ticket and mark it for 1.22.2, we can deal with it in January 2024.
